src/sbvr-api/sbvr-utils.ts

```diff
@@ -1024,7 +1024,27 @@ const getIdField = (
 	>,
 ) => getAbstractSqlModel(request).tables[resolveSynonym(request)].idField;
 
-export const getAffectedIds = async ({
+export const getAffectedIds = async (args: {
+	req: HookReq;
+	request: HookRequest;
+	tx: Db.Tx;
+}): Promise<number[]> => {
+	const { request } = args;
+	if (request.affectedIds) {
+		return request.affectedIds;
+	}
+	// We store the affected ids promise so we only have to fetch them once per request
+	if (request.pendingAffectedIds == null) {
+		request.pendingAffectedIds = $getAffectedIds(args);
+	}
+
+	const affectedIds = await request.pendingAffectedIds;
```

Comment:
Allows us to avoid having a `!` in the `return affectedIds;`, which we would have to do if we did `return request.affectedIds;`.
Feels cleaner as well imo :)

Reply:
Actually @Page- , while running the api tests I realized that the `request.affectedIds` on `master` is assigned on the re-assigned `request` object (`request = await uriParser.parseOData(/*...*/)`.)
Is that correct?
See: https://github.com/balena-io/pinejs/blob/5c62bbda68129809635e8a040ab81c6c992df71c/src/sbvr-api/sbvr-utils.ts#L1045-L1048